rust/src/state_machine/coordinator.rs

```diff
@@ -0,0 +1,112 @@
+use std::collections::HashMap;
+
+use sodiumoxide::{self, crypto::box_, randombytes::randombytes};
+
+use crate::{
+    crypto::{ByteObject, KeyPair},
+    mask::{MaskConfig, MaskObject},
+    state_machine::events::{EventPublisher, EventSubscriber},
+};
+
+pub type RoundId = u64;
+
+#[derive(Debug, Clone)]
+pub struct RoundParameters {
+    pub id: RoundId,
```

Comment:
### Re: `RoundId

Agree. I'll remove it

### Re: not broadcasting all the round parameters in the Idle phase

With the latest changes, we now broadcast the round parameters in the Idle phase. There's is not risk of participants getting a head start, since all the parameters are  published at once.